fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1343,6 +1343,16 @@ Future< Standalone<VectorRef<const char*> >> ReadYourWritesTransaction::getAddre
 	return result;
 }
 
+Future<int64_t> ReadYourWritesTransaction::getEstimatedRangeSizeBytes(const KeyRangeRef& keys) {
+	if(checkUsedDuringCommit()) {
+		throw used_during_commit();
+	}
+	if( resetPromise.isSet() )
+		return resetPromise.getFuture().getError();
+
+	return map(tr.getStorageMetrics(keys, -1), [](const StorageMetrics& m) { return m.bytes; });
```

Comment:
I think this should also be made to fail early if `resetPromise` gets set while the operation is outstanding. It looks like `getAddressesForKey` above is using `waitOrError` to accomplish this, which I think we could do here too.